x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import type { FC } from 'react';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiResizeObserver } from '@elastic/eui';
+import type { CoreStart } from '@kbn/core-lifecycle-browser';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { KibanaRenderContextProvider } from '@kbn/react-kibana-context-render';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import type { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import { DatePickerContextProvider, type DatePickerDependencies } from '@kbn/ml-date-picker';
+import type { TimeRangeBounds } from '@kbn/ml-time-buckets';
+import usePrevious from 'react-use/lib/usePrevious';
+import { tz } from 'moment';
+import { pick, throttle } from 'lodash';
+import type { MlDependencies } from '../../application/app';
+import { TimeSeriesExplorerEmbeddableChart } from '../../application/timeseriesexplorer/timeseriesexplorer_embeddable_chart';
+import { APP_STATE_ACTION } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import type { SingleMetricViewerServices, MlEntity } from '../../embeddables/types';
+import './_index.scss';
+
+const containerPadding = 10;
+const minElemAndChartDiff = 20;
+const RESIZE_THROTTLE_TIME_MS = 500;
+interface AppStateZoom {
+  from?: string;
+  to?: string;
+}
+
+const errorMessage = i18n.translate('xpack.ml.singleMetricViewerEmbeddable.errorMessage"', {
+  defaultMessage: 'Unable to load the ML single metric viewer data',
+});
+
+export type SingleMetricViewerSharedComponent = FC<SingleMetricViewerProps>;
+
+/**
+ * Only used to initialize internally
+ */
+export type SingleMetricViewerPropsWithDeps = SingleMetricViewerProps & {
+  coreStart: CoreStart;
+  pluginStart: MlDependencies;
+  mlServices: SingleMetricViewerServices;
+};
+
+export interface SingleMetricViewerProps {
+  bounds?: TimeRangeBounds;
+  selectedEntities?: MlEntity;
+  selectedDetectorIndex?: number;
+  functionDescription?: string;
+  selectedJobId: string | undefined;
+  /**
+   * Last reload request time, can be used for manual reload
+   */
+  lastRefresh?: number;
+  onLoading?: (isLoading: boolean) => void; // TODO: callback
```

Comment:
is it TODO still relevant?

Reply:
Removed in 6c61014b8a7ef4a3167a642e5f72f1938ea25c50